src/sentry/sentry_metrics/querying/data/execution.py

```diff
@@ -324,9 +376,11 @@ def align_series_to_totals(self) -> "QueryResult":
 
 
 class QueryExecutor:
-    def __init__(self, organization: Organization, referrer: str):
+    def __init__(self, organization: Organization, projects: Sequence[Project], referrer: str):
```

Comment:
nit: can we get the org from the list of projects since we don't allow cross-org queries?

Reply:
The reason for why we have the organization as separate param, is to keep the implementation consistent with the organization supplied by the endpoint.

It's mostly a stylistical choice but we could definitely make a function that extracts one organization from upstream (and throws an error in case there are multiple) and sets that across the components of the api.